src/hotspot/cpu/aarch64/interpreterRT_aarch64.cpp

```diff
@@ -51,6 +78,39 @@ InterpreterRuntime::SignatureHandlerGenerator::SignatureHandlerGenerator(
   _stack_offset = 0;
 }
 
+// On macos/aarch64 native stack is packed, int/float are using only 4 bytes
+// on stack. Natural alignment for types are still in place,
+// for example double/long should be 8 bytes alligned
+
```

Comment:
This comment is a bit confusing because it's no longer #ifdef APPLE. Better move it up to Line 41.